### ISSUE TITLE
some place where there isn't type safety throws error with .type of undefined

### ISSUE THREAD

**Issue by aretecode** [mobxjs/mobx-state-tree#469]
<img width="992" alt="screen shot 2017-10-14 at 3 29 09 am" src="https://user-images.githubusercontent.com/4022631/31574953-039a21c6-b090-11e7-9343-0f5022a84740.png">

**Comment by mattiamanzati** [comment]
Uhm, can you please provide a little bit more of code to reproduce please? :)

**Comment by aretecode** [comment]
```js

const Tree = types
  .model({
    type: 'eh',
    label: 'eh',
  })
  .named('eh')

const state = Tree.create()
Tree.applySnapshot(state, {
  type: 'oh',
})
```

this being the hint
<img width="530" alt="screen shot 2017-10-14 at 3 34 41 am" src="https://user-images.githubusercontent.com/4022631/31574988-a8bad36c-b090-11e7-85b9-9e084664ee2d.png">

**Comment by mattiamanzati** [comment]
Yeah, that method is internal and should be hidden indeed, the correct way is importing and calling :)

**Comment by mattiamanzati** [comment]
Idea: Store "TypeNodeAdministration" over a "$type" property in the type, so we can "hide" it from the developer

**Comment by xaviergonz** [comment]
@mattiamanzati I see that types now have a $tree variable, can this issue be closed?
